app/autotranslate/server/googleTranslate.js

```diff
@@ -0,0 +1,188 @@
+/**
+ * @author Vigneshwaran Odayappan <vickyokrm@gmail.com>
+ */
+
+import { TAPi18n } from 'meteor/rocketchat:tap-i18n';
+import { HTTP } from 'meteor/http';
+import _ from 'underscore';
+
+import { AutoTranslate,	TranslationProviderRegistry } from './autotranslate';
+import { SystemLogger } from '../../logger/server';
+import { settings } from '../../settings';
+
+/**
+ * Represents google translate class
+ * @class
+ * @augments AutoTranslate
+ */
+class GoogleAutoTranslate extends AutoTranslate {
+	/**
+	 * setup api reference to Google translate to be used as message translation provider.
+	 * @constructor
+	 */
+	constructor() {
+		super();
+		this.name = 'google-translate';
+		this.apiEndPointUrl = 'https://translation.googleapis.com/language/translate/v2';
+		// Get the service provide API key.
+		settings.get('AutoTranslate_DeepLAPIKey', (key, value) => {
```

Comment:
Wait isn’t this the wrong key to be grabbing in Google section?

Reply:
I'd think it should be `AutoTranslate_GoogleAPIKey `